homework_week_14/src/main/java/homework/Main.java

```diff
@@ -0,0 +1,168 @@
+                .forEach(index -> randomIntegers.add((int) (Math.random() * 100)));
+
+        System.out.print("\n(5) The amount of random numbers: ");
+        System.out.println(randomIntegers.size());
+
+        System.out.println("The exponential of the random numbers: ");
+        randomIntegers.stream()
+                .map(Math::exp)
+                .forEach(System.out::println);
+
+
+
+        /*
+         * 6) Create a map of "n" (K,V) elements
+         * and print "how many" elements have value over 10 (the key is of your choice)
+         */
+        Map<Integer, Double> cubes = new LinkedHashMap<>();
+
+        int n = 5;
+        IntStream.range(0, n).forEach(index -> cubes.put(index, Math.pow(index, 3)));
+
+        System.out.print("\n(6) The number of elements over 10 when n=5 is: "
+                + cubes.values()
+                .stream()
+                .filter(value -> value > 10)
+                .count());
+
+
+
+        /*
+         * 7) Sort the above Set<String> (used for max) in reverse order
+         */
+        System.out.println("\n\n(7) Random strings reverse ordered: ");
+        randomStrings.stream()
+                .sorted(Comparator.reverseOrder())
+                .forEach(System.out::println);
+
+
+
+        /*
+         * 8) Sort the above List of custom objects (used for filter) in an order you consider
+         */
+        System.out.println("\n(8) Car list sorted by horsepower: ");
+        carList.stream()
+                .sorted(Comparator.comparing(Car::getHorsePower))
+                .forEach(System.out::println);
+
+
+
+        /*
+         * 9) Check if any of your instances "match" a condition based on an Integer field (of your choice).
+         * What does it return ? Print it out.
+         */
+        System.out.print("\n\n(9) Are there any cars made in 2020? : ");
+        System.out.print(carList.stream().anyMatch(car -> car.getYear() == 2020) ? "Yes" : "No");
+
+
+
+        /*
+         * 10) What does Optional represent ? Explain through an example on your custom object
+         *  -> Wrap an existing instance
+         *  -> Wrap a null
+         *  -> Check value using ifPresent or isPresent
+         */
+        System.out.println("\n\n(10) Electric cars: ");
+        carList.stream()
+                .filter(car -> car.getIsElectric().equals(Optional.of(true)))
+                .forEach(System.out::println);
+        System.out.println("Cars with missing info: ");
+        carList.stream()
+                .filter(car -> !car.getIsElectric().isPresent())
+                .forEach(System.out::println);
+
+
+
+        /*
+         * 11) Fastest way to find the shortest String in a List
+         * (take the 5 random Strings generated above and find the shortest one).
+         */
+        System.out.println("\n\n(11) Shortest string in the list: "
+                + randomStrings.stream()
+                .min(Comparator.comparingInt(String::length))
```

Comment:
It's not actually the fastest :), it depends on how many elements there are generated, what other option is there ? (discussed in class)